prow/github/client.go

```diff
@@ -3169,7 +3169,19 @@ func (c *client) IsMergeable(org, repo string, number int, SHA string) (bool, er
 			return false, errors.New("pull request was merged while checking mergeability")
 		}
 		if pr.Mergable != nil {
-			return *pr.Mergable, nil
+			// In certain cases, the mergeable field is lying.
+			switch pr.MergeableState {
+			case MergeableStateBehind:
+				fallthrough
+			case MergeableStateBlocked:
+				fallthrough
+			case MergeableStateDraft:
+				fallthrough
+			case MergeableStateUnknown:
+				return false, fmt.Errorf("pull request is in unmergeable state %v", pr.MergeableState)
```

Comment:
```suggestion
			// In certain cases, the mergeable field is lying.
			switch pr.MergeableState {
			case MergeableStateBehind, MergeableStateBlocked, MergeableStateDraft, MergeableStateUnknown:
				return false, fmt.Errorf("pull request is in unmergeable state %v", pr.MergeableState)
```
Any reason why can't just merge the cases?

Reply:
👍